### PR TITLE
feat(ats): [FSSDK-8690] fix configurable ODP request timeouts

### PR DESCRIPTION
**Pull request by jaeopt**
## Summary
Support configurable timeouts for ODP segment and event requests.
Use default timeout of 10 secs if not configured.

## Test plan
- Add unit tests for configured and default timeout values

## Issues
- [FSSDK-8690](https://jira.sso.episerver.net/browse/FSSDK-8690)

### CODE REVIEW

File: packages/optimizely-sdk/tests/odpManager.browser.spec.ts [hunk 1]
```diff
@@ -336,6 +336,13 @@ describe('OdpManager', () => {
       expect(browserOdpManager.segmentManager.odpSegmentApiManager.requestHandler.timeout).toBe(4000);
     });
 
+    it('Browser default Segments API Request Handler timeout should be used when odpOptions does not include segmentsApiTimeout', () => {
+      const browserOdpManager = new BrowserOdpManager({});
+
+      // @ts-ignore
+      expect(browserOdpManager.segmentManager.odpSegmentApiManager.requestHandler.timeout).toBe(10000);
```
Comment (yasirfolio3) [inline packages/optimizely-sdk/tests/odpManager.browser.spec.ts:343]: can we use `REQUEST_TIMEOUT_ODP_SEGMENTS_MS ` here? same for the other test.

**Review by yasirfolio3 — APPROVED** [review]
Just a minor suggestion, LGTM otherwise.

File: packages/optimizely-sdk/tests/odpManager.browser.spec.ts (hunk 1 above)
Comment (jaeopt) [reply packages/optimizely-sdk/tests/odpManager.browser.spec.ts:343]: I agree that comparing against the variable will improve test reliability when adjusting the timeout. In this case, I intentionally check against the number to protect unintended constant changes :)

**Review by opti-jnguyen — APPROVED** [review]
LGTM!

Merging master back in to be synced with the `jae/8793-odp-flush` changes and will force merge this in afterwards.

**Pull request merged**